### ISSUE TITLE
Integration test for logs tail/write keeps failing silently

### ISSUE THREAD

**Issue by matej-g** [observatorium/api#153]
It seems that this particular test keeps failing permanently, without an explicit failure message. This applies to running it both locally and on CircleCI (see e.g. recent run here - https://app.circleci.com/pipelines/github/observatorium/api/87/workflows/d8c59c8d-f3c8-4b45-b0f3-50700da87d7a/jobs/357 - but since it exits with code `0`, I assume that is the reason is shows up as green in CI).

Relevant excerpt from output (locally run `make test-integration`):
```
-------------------------------------------
- Logs Read/Write tests: OK               -
-------------------------------------------
-------------------------------------------
- Logs Tail/Write tests                   -
-------------------------------------------
level=debug name=observatorium ts=2021-07-13T10:31:43.501425403Z caller=instrumentation.go:36 request=mgera.remote.csb/JKeZgC35Pg-000080 proto=HTTP/2.0 method=POST status=404 content=application/json path=/api/logs/v1/test-oidc/api/v1/push duration=678.177µs bytes=19
Can't use SSL_get_servername
depth=1 CN = observatorium
verify return:1
depth=0 CN = localhost
verify return:1
level=debug name=observatorium ts=2021-07-13T10:31:43.508231948Z caller=instrumentation.go:36 request=mgera.remote.csb/JKeZgC35Pg-000081 proto=HTTP/1.1 method=GET status=404 content= path=/api/logs/v1/test-oidc/api/v1/tail duration=146.55µs bytes=19
websocat: WebSocketError: Received unexpected status code
websocat: error running
level=info name=observatorium ts=2021-07-13T10:31:43.509220823Z caller=main.go:354 msg="caught interrupt"                                                                                                         
level=info name=observatorium ts=2021-07-13T10:31:43.509247696Z caller=main.go:598 msg="shutting down the HTTP server"                                                                                            
{"level":"info","msg":"Shutting down...","time":"2021-07-13T12:31:43+02:00"}
level=error ts=2021-07-13T10:31:43.509597488Z caller=transfer.go:192 msg="transfer failed" err="cannot find ingester to transfer chunks to: no pending ingesters"                                                 
{"level":"info","msg":"Server shutdown.","time":"2021-07-13T12:31:43+02:00"}
```

**Comment by periklis** [comment]
Addressed by #154

**Comment by matej-g** [comment]
@periklis Did it provide the intended fix? I still see it failing locally and on PRs run as well (https://app.circleci.com/pipelines/github/observatorium/api/89/workflows/f4366180-6190-4aaa-a6cc-a3b9c50d48b3/jobs/364)
